pkg/api/dirs.go

```diff
@@ -128,7 +128,15 @@ func storeDir(ctx context.Context, reader io.ReadCloser, s storage.Storer, mode
 			return swarm.ZeroAddress, fmt.Errorf("read tar stream: %w", err)
 		}
 
-		filePath := fileHeader.Name
+		filePath := filepath.Clean(fileHeader.Name)
+
+		if filePath == "." {
+			logger.Warningf("skipping file upload empty path")
+			continue
+		}
+
+		// always use Unix path separator
+		filePath = strings.ReplaceAll(filePath, "\\", "/")
```

Comment:
minor comment, that `\\` is a valid POSIX filename. there's no other solution here though, since we cannot know on which operating system the tarball was created, and I don't know if a tarball contained on a windows machine writes the metadata header with POSIX compatible filenames